tests/functional/ui/test_odf_storage_consumption_trend.py

```diff
@@ -0,0 +1,314 @@
+import logging
+import math
+import re
+import time
+
+from ocs_ci.framework import config
+from ocs_ci.framework.testlib import (
+    ManageTest,
+    tier2,
+    ui,
+    ignore_leftovers,
+    black_squad,
+    skipif_ocp_version,
+    skipif_ocs_version,
+)
+from ocs_ci.helpers import helpers
+from ocs_ci.helpers.osd_resize import basic_resize_osd
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.cluster import get_used_and_total_capacity_in_gibibytes
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.resources.pod import (
+    get_age_of_cluster_in_days,
+    get_mgr_pods,
+    get_ceph_tools_pod,
+    delete_pods,
+    get_prometheus_pods,
+)
+from ocs_ci.ocs.resources.storage_cluster import get_storage_size
+from ocs_ci.ocs.ui.validation_ui import ValidationUI
+
+logger = logging.getLogger(__name__)
+
+
+POD_OBJ = OCP(kind=constants.POD, namespace=config.ENV_DATA["cluster_namespace"])
+
+
+@tier2
+@ui
+@ignore_leftovers
+@black_squad
+@skipif_ocp_version("<4.17")
+@skipif_ocs_version("<4.17")
+class TestConsumptionTrendUI(ManageTest):
+    """
+    Few test cases in this class requires manual calculations for 'average of storage consumption'
+    and 'estimated days until full'. We derived values from UI, also calculated manually and compared them.
+
+    Note:
+    The values obtained from manual calculations and the UI may exhibit slight differences
+     due to back filling data or background activities.
+    To accommodate this, a tolerance of 10% difference between the manual and UI values is allowed.
+
+    For example, if the manual calculation yields a value of 10 and the UI shows 11 as the average,
+    the test will still pass.
+    However, if the UI value is 12 or greater, the test will fail due to exceeding the 10% tolerance.
+
+    """
+
+    def get_est_days_from_ui(self):
+        """
+        Get the value of 'Estimated days until full' from the UI
+
+        Returns:
+            est_days: (int)
+
+        """
+        validation_ui_obj = ValidationUI()
+        collected_tpl_of_days_and_avg = (
+            validation_ui_obj.odf_storagesystems_consumption_trend()
+        )
+        est_days = re.search(r"\d+", collected_tpl_of_days_and_avg[0]).group()
+        logger.info(f"'Estimated days until full' from the UI : {est_days}")
+        return int(est_days)
+
+    def get_avg_consumption_from_ui(self):
+        """
+        Get the value of 'Average storage consumption' from the UI
+
+        Returns:
+            average: (float)
+
+        """
+        validation_ui_obj = ValidationUI()
+        collected_tpl_of_days_and_avg = (
+            validation_ui_obj.odf_storagesystems_consumption_trend()
+        )
+        average = float(
+            re.search(r"-?\d+\.*\d*", collected_tpl_of_days_and_avg[1]).group()
+        )
+        logger.info(f"'Average of storage consumption per day' from the UI : {average}")
+        return average
+
```

Comment:
this utility can be reused. 
pls move `calculate_est_days_and_average_manually` to ValidationUI instead of creating ValidationUI instance in every function

Reply:
Average consumption will change for every second and it depends on the space we are filling in the cluster. To get the accurate values, we need to call this function in every test function on need basis.

moved `calculate_est_days_and_average_manually` to ValidationUI